File: scripts/parse_trivy.js
const fs = require('fs');

const raw = fs.readFileSync('D:\\LiveKit-Meeting\\trivy-backend.json', 'utf8');
const data = JSON.parse(raw);

data.Results?.forEach(result => {
    result.Vulnerabilities?.forEach(vuln => {
        if (vuln.Severity === 'HIGH') {
            console.log(`CVE: ${vuln.VulnerabilityID}`);
            console.log(`Package: ${vuln.PkgName}`);
            console.log(`Installed: ${vuln.InstalledVersion}`);
            console.log(`Fixed: ${vuln.FixedVersion}`);
            console.log(`Target: ${result.Target}`);
            console.log('-'.repeat(40));
        }
    });
});

File: scripts/parse_trivy.py
import json
import sys

def parse_trivy():
    with open(r'D:\LiveKit-Meeting\reports\trivy-backend-new.json', 'r', encoding='utf-8') as f:
        data = json.load(f)
    
    for result in data.get("Results", []):
        for vuln in result.get("Vulnerabilities", []):
            if vuln.get("Severity") == "HIGH":
                print(f"CVE: {vuln.get('VulnerabilityID')}")
                print(f"Package: {vuln.get('PkgName')}")
                print(f"Installed: {vuln.get('InstalledVersion')}")
                print(f"Fixed: {vuln.get('FixedVersion')}")
                print(f"Source: {vuln.get('DataSource', {}).get('Name')}")
                print(f"Layer: {result.get('Target')}")
                print("-" * 40)

if __name__ == "__main__":
    parse_trivy()
